### DISCUSSION TITLE
Is there a good way to import saved Eventlogs from Windows?

### DISCUSSION THREAD

**Discussion by zaphode** [Security-Onion-Solutions/securityonion#2194] [Unsupported Versions] [answered]
I built an import only analyst SO and am trying to figure out the best way to import stored EventLogs and syslogs. I took the Security Onion Fundamentals class a few weeks back and in there we used a docker container (so-evtx2es) to import .evtx files. I cant seem to find that in the import build (It seems like that should be there). Is there something I am missing or could someone point me to the docs on how to do this? 
I am assuming that the so-evtx2es container is a port of evtx2es that is optimized to work in SO.

**Comment by defensivedepth** [comment, marked as answer]
The container used in the SO Fundamentals class is a fork of evtx2es with some additional functionality (ability to time-shift all logs by a certain timeframe, use ES ingest node parsing, etc).

That has not been integrated into Import Node yet, though we do have it planned. I have created an issue to track this - https://github.com/Security-Onion-Solutions/securityonion/issues/2206 

The vast majority of work has been done, but there are some additional things that need to be figured out. I'm interested in hearing your use case - Would you want the evtx logs to be imported with the original timestamps, or shifted in some way? Also, would you expect to be able to run all currently-active Playbook Plays against the imported logs?

**Reply by zaphode** [reply]
- Would you want the evtx logs to be imported with the original timestamps, or shifted in some way? Original timestamps
- would you expect to be able to run all currently-active Playbook Plays against the imported logs? I guess i assumed the Plays were run against the post ingested log data but yes it would be great to be able run plays against the logs.

**Reply by defensivedepth** [reply]
Thanks much! Hope to get this done soon-ish!

**Reply by zaphode** [reply]
One more question. I can get the saved event logs into Elasticsearch using the base evtx2es app. It looks like it is missing a few of the metadata fields (event.catagory.keyword, event.dataset.keyword, etc...) that are present when importing the logs live from Wazuh that allow the Sysmon Dashboard to find them. Are these values generated by an ingest parser or do these need to be added to the evtx2es.py script that sends them to Elasticsearch?

**Reply by defensivedepth** [reply]
If you use this branch (https://github.com/defensivedepth/evtx2es/tree/feature/so), it adds the ability to specify an ingest pipeline, which we already have in SO by default - `import.wel` 

(https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/elasticsearch/files/ingest/import.wel)

That should parse everything out correctly, but YMMV until we get it officially supported in the `Import` node

https://github.com/defensivedepth/evtx2es/blob/feature/so/src/evtx2es.py#L255
